### ISSUE TITLE
About the accuracy of the trained model

### ISSUE THREAD

**Issue by longzaitianguo** [inspirehep/magpie#106]
I trained the model on a multi label classification dataset, which has about 400,000 instances and 500 labels in total. When the model is training, the accuracy of the train data and test data reach almost 1 at the first epoch. In total, I trained the model for 50 epochs.

However, when I test the trained model by myself, no instances are predicted correctly. I take the first two labels for each instance, and none of them is a correct label. The performance is rather poor.

Any intuition about this strange accuracy?

**Comment by jstypka** [comment]
So accuracy is not a great metric to look at if you have a very sparse target vector. You mentioned that you have 500 labels, so if usually one of them is correct, the model would get 99.8% accuracy just by predicting only zeros. You probably should look at the validation loss (`val_loss`) and see if it's decreasing and choose an appropriate number of epochs that way. I will implement (in an undefined future) F1 score in place of accuracy, so it looks a little better.

**Comment by tarvos21** [comment]
@jstypka Where can I change the value of `val_loss` of CNN model in Magpie?

**Comment by longzaitianguo** [comment]
@jstypka How can I change the value of SAMPLE_LENGTH, since the length of my instances is larger than the default value : 200. I do not want to build the package again once I change the value of SAMPLE_LENGTH in the config.py.

**Comment by jstypka** [comment]
@tarvos21 this value is not set anywhere. It is a metric computed by Magpie to describe how accurate it is, you don't set it anywhere.

@longzaitianguo I'm afraid `config.py` is the only place where you can change it. You can install the package with the -e flag, so that you don't have to reinstall it after every change e.g. `pip install -e ...`

**Comment by tarvos21** [comment]
@jstypka Thanks for your patience very much.

**Comment by longzaitianguo** [comment]
@jstypka Do you have any experience using Magpie on a very sparse target vector? Just like the 500 labels I mentioned. I run 200 epochs and the accuracy is still very poor. I returned top 5 labels, and the precision and recall are both below 1%.

**Comment by jstypka** [comment]
@longzaitianguo that depends on your training set. 500 labels should definitely be manageable, we run it for 10k labels at some point - it just demands a huge training dataset, but Magpie scales to these sizes. Also make sure you don't overfit after running 200 epochs, that's quite a lot.

**Comment by longzaitianguo** [comment]
@jstypka Thanks a lot for your comment.
